central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -54,23 +90,162 @@ func (m *managerImpl) ProcessComplianceOperatorInfo(ctx context.Context, complia
 	return m.integrationDS.UpdateComplianceIntegration(ctx, complianceIntegration)
 }
 
-func (m *managerImpl) ProcessScanRequest(_ context.Context, _ interface{}) error {
-	//TODO:
-	// 1. Upsert to database
-	// 2. Push request to Sensor
-	panic("implement me")
+// ProcessScanRequest processes a request to apply a compliance scan configuration to one or more Sensors.
+func (m *managerImpl) ProcessScanRequest(ctx context.Context, scanRequest *storage.ComplianceOperatorScanConfigurationV2, clusters []string) (*storage.ComplianceOperatorScanConfigurationV2, error) {
+	if !features.ComplianceEnhancements.Enabled() {
+		return nil, errors.Errorf("Compliance is disabled. Cannot process scan request: %q", scanRequest.GetScanName())
+	}
+
+	// Convert and validate schedule
+	// TODO(ROX-19818):  remove default schedule
+	// For MVP a schedule is required.  If one is not present use daily at midnight.  As it evolves
+	// it may be as simple as no schedule means a one time scan.
+	cron := defaultScanSchedule
+	var err error
+	if scanRequest.GetSchedule() != nil {
+		cron, err = schedule.ConvertToCronTab(scanRequest.GetSchedule())
+		if err != nil {
+			err = errors.Wrapf(err, "Unable to convert schedule for scan configuration named %q to cron.", scanRequest.GetScanName())
+			log.Error(err)
+			return nil, err
+		}
+		cronValidator := gronx.New()
+		if !cronValidator.IsValid(cron) {
+			err = errors.Errorf("Schedule for scan configuration named %q is invalid.", scanRequest.GetScanName())
+			log.Error(err)
+			return nil, err
+		}
+	}
+
+	// Check if scan configuration already exists.
+	found, err := m.scanSettingDS.GetScanConfigurationExists(ctx, scanRequest.GetScanName())
```

Comment:
What was the motivation to choose `GetScanConfigurationExists` over `ScanConfigurationExists`?

Reply:
no motivation.  I copied a method signature and stuck exists on the end in a rush.  I will update it